ansible_3par_docker_plugin/tasks/create_conf_file.yml

```diff
@@ -0,0 +1,223 @@
+    become: yes
+    when:  item.1.value in INVENTORY[item.0].keys()
+
+  - name: Populate etcd IP in hpe.conf
+    ini_file:
+      path: /etc/hpedockerplugin/hpe.conf
+      section: "{{ item[0] }}"
+      option: "{{ item[1]['option'] }}"
+      value: "{{ item[1]['value'] }}"
+      no_extra_spaces: true
+    with_nested:
+      - "{{ INVENTORY.keys()}}"
+      - [{ option: 'host_etcd_ip_address', value: "{{ hostvars[inventory_hostname]['ansible_default_ipv4']['address'] }}" }]
+    become: yes
+
+  - name: Populate WSAPI URL in hpe.conf
+    ini_file:
+      path: /etc/hpedockerplugin/hpe.conf
+      section: "{{ item[0] }}"
+      option: "{{ item[1]['option'] }}"
+      value: "{{ 'https://' + INVENTORY[item[0]][item[1].value] + ':8080/api/v1' }}"
+      no_extra_spaces: true
+    with_nested:
+      - "{{ INVENTORY.keys()}}"
+      - [{ option: 'hpe3par_api_url', value: 'hpe3par_ip' }]
+    become: yes
+
+  - name: Populate iscsi IPs in hpe.conf
+    ini_file:
+      path: /etc/hpedockerplugin/hpe.conf
+      section: "{{ item['item'] }}"
+      option: 'hpe3par_iscsi_ips'
+      value: "{{ ','.join(item['stdout_lines']) }}"
+      no_extra_spaces: true
+    with_items:
+      - "{{ primary_iscsi_ports.results }}"
+    become: yes
+    when: INVENTORY[item['item']]['hpedockerplugin_driver'] == 'hpedockerplugin.hpe.hpe_3par_iscsi.HPE3PARISCSIDriver'
+
+  - name: Get the encrypted passwords
+    shell: hpe3parencryptor --backend {{ item[0] }} -a {{INVENTORY[item[0]]['encryptor_key']}} "{{ INVENTORY[item[0]][(item[1].value)] }}"
+    ignore_errors: yes
+    register: passwords
+    with_nested:
+      - "{{ INVENTORY.keys()}}"
+      - [{ option: 'hpe3par_password', value: 'hpe3par_password' }, { option: 'san_password', value: 'hpe3par_password' }]
+    when: INVENTORY[item[0]]['encryptor_key'] is defined
+
+  - name: Populate the encrypted passwords in hpe.conf
+    ini_file:
+      path: /etc/hpedockerplugin/hpe.conf
+      section: "{{ passwords['results'][item | int]['item'][0] }}"
+      option:  "{{ passwords['results'][item | int]['item'][1]['option'] }}"
+      value: "{{ passwords['results'][item | int]['stdout_lines'][0].split('password: ')[1] }} "
+      no_extra_spaces: true
+    with_items:
+      - "{{ lookup('sequence','start=0 end='+((INVENTORY.keys() | count)*2 -1) |string,wantlist=True) }}"
+    become: yes
+    when:  passwords['results'][item | int]['stdout_lines'] is defined
+    ignore_errors: yes
+
+  - name: Wait 15 seconds
+    wait_for: timeout=10
+
+  - name: Get the encrypted passwords
+    shell: hpe3parencryptor --backend {{ item[0] }} -a {{INVENTORY[item[0]]['encryptor_key']}} "{{ INVENTORY[item[0]][(item[1].value)] }}"
+    ignore_errors: yes
+    register: passwords
+    with_nested:
+      - "{{ INVENTORY.keys()}}"
+      - [{ option: 'hpe3par_password', value: 'hpe3par_password' }, { option: 'san_password', value: 'hpe3par_password' }]
+    when: INVENTORY[item[0]]['encryptor_key'] is defined
```

Comment:
Shouldn't we also check for None in such condition where we validate if the inventory item is defined?
For example:
when: INVENTORY[item[0]]['encryptor_key'] is defined and INVENTORY[item[0]]['encryptor_key'] != None
Will this be a good check?

Reply:
Defined should check for existence of the key. So the none check is covered with defined. Also, we expect at least one backend to exist in the configuration so checking for the existence of item[0] which refers to the first loop for inventory keys may not be required.